VagrantProvision.sh

```diff
@@ -20,9 +20,19 @@ sudo ntpd -gq
 sudo service ntp start
 
 echo "### Installing Java 8..."
-sudo wget --quiet --no-check-certificate --no-cookies --header "Cookie: oraclelicense=accept-securebackup-cookie" http://download.oracle.com/otn-pub/java/jdk/8u112-b15/jdk-8u112-linux-x64.tar.gz -P /tmp
-sudo tar -xvzf /tmp/jdk-8u112-linux-x64.tar.gz --directory=/tmp >/dev/null
 
+# jdk-8u112-linux-x64.tar.gz's official checksums:
+#    sha256: 777bd7d5268408a5a94f5e366c2e43e720c6ce4fe8c59d9a71e2961e50d774a5
+#    md5: de9b7a90f0f5a13cfcaa3b01451d0337
+echo "de9b7a90f0f5a13cfcaa3b01451d0337  /tmp/jdk-8u112-linux-x64.tar.gz" > /tmp/jdk.md5
+
+if [ ! -f /tmp/jdk-8u112-linux-x64.tar.gz ] || ! md5sum -c /tmp/jdk.md5; then
```

Comment:
Is the md5sum command here to check that the existing tarball is not corrupt in any way?